.clang-format

```diff
@@ -0,0 +1,104 @@
+AllowShortFunctionsOnASingleLine: Empty
+# Do not allow short if statements on a single line.
+AllowShortIfStatementsOnASingleLine: Never
+# Do not allow short loops on a single line.
+AllowShortLoopsOnASingleLine: false
+# Do not pack function call arguments together.
+# A function call’s arguments will either be all on the same line or will have one line each.
+BinPackArguments: false
+# Do not pack function parameters together.
+# A function declaration’s or function definition’s parameters will either all be on the same line
+# or will have one line each.
+BinPackParameters: false
+# Add a space after the colon in bit fields.
+BitFieldColonSpacing: After
+# Break before binary operators that are not assignments.
+# E.g. break before "+", but not before "+=".
+BreakBeforeBinaryOperators: NonAssignment
+# Attach opening braces to the same line as the statement or declaration it belongs to.
+BreakBeforeBraces: Attach
+# Break string literals if they are too long.
+BreakStringLiterals: true
+# Maximum column width before code will be wrapped or formatted.
+ColumnLimit: 110
+# Indentation width for constructor initializers.
+ConstructorInitializerIndentWidth: 4
+# Indentation width for continuation lines.
+ContinuationIndentWidth: 4
+# Format braced lists as best suited for C++11 braced lists.
+Cpp11BracedListStyle: true
+# Use a fixed alignment for all pointers.
+DerivePointerAlignment: false
+# Indent case blocks in switch statements.
+IndentCaseBlocks: true
+# Indent case labels in switch statements.
+IndentCaseLabels: true
+# Standard indentation width.
+IndentWidth: 4
+# Do not indent if a function definition or declaration is wrapped after the type.
+IndentWrappedFunctionNames: false
+# Insert a newline at end of file if missing.
+InsertNewlineAtEOF: true
+# Maximum number of consecutive empty lines to keep.
+MaxEmptyLinesToKeep: 2
+# Do not indent namespaces.
+NamespaceIndentation: None
+# Put all constructor initializers on the current line if they fit. Otherwise, put each one on its own line.
+PackConstructorInitializers: CurrentLine
+# Penalty for indentation using spaces rather than tabs.
+PenaltyIndentedWhitespace: 0
+# Pointer alignment (Left or Right).
+PointerAlignment: Left
+# Separate definition blocks (Always, Never, or Leave).
+SeparateDefinitionBlocks: Always
+# Number of spaces before trailing comments.
+SpacesBeforeTrailingComments: 1
+# Do not add a space after a C-style cast.
+SpaceAfterCStyleCast: false
+# Do not add a space after the template keyword.
+SpaceAfterTemplateKeyword: false
+# Sort includes (CaseSensitive, CaseInsensitive, or None).
+SortIncludes: CaseInsensitive
+# C++ standard to be followed (e.g., c++98, c++11, c++14, c++17, c++20).
+Standard: c++20
+# Add a space before opening parentheses in control statements.
+SpaceBeforeParens: ControlStatements
+# Do not add a space before square brackets.
+SpaceBeforeSquareBrackets: false
+# Width of a tab character.
+TabWidth: 4
```

Comment:
I wonder if we should base things around 2 spaces per tab rather than 4? If we were using spaces instead of tabs, I'd be moving to 2.

Reply:
AH, that's fine with me.